src/ansys/dpf/core/mesh_info.py

```diff
@@ -0,0 +1,166 @@
+"""
+MeshInfo
+==========
+"""
+import ansys.dpf.core
+from ansys.dpf.core import server as server_module
+
+
+class MeshInfo:
+    """Represents the mesh information.
+
+    This class describes the available mesh information.
+
+    Parameters
+    ----------
+    mesh_info : ctypes.c_void_p, ansys.grpc.dpf.mesh_info_pb2.MeshInfo message
+
+     server : ansys.dpf.core.server, optional
+        Server with the channel connected to the remote or local instance.
+        The default is ``None``, in which case an attempt is made to use the
+        global server.
+
+    Examples
+    --------
+    Explore the mesh info from the model
+
+    >>> from ansys.dpf import core as dpf
+    >>> from ansys.dpf.core import examples
+    >>> fluent = examples.download_fluent_axial_comp()
+    >>> model = dpf.Model(fluent)
+    >>> mesh_info = model.metadata.mesh_info_provider # printable mesh_info
+
+    """
+
+    def __init__(self, gdc=None, mesh_info=None, server=None):
+        """Initialize with a MeshInfo message"""
+        # ############################
+        # step 1: get server
+        self._server = server_module.get_or_create_server(server)
+
+        try:
+            if gdc is None and mesh_info is None:
+                self._gdc = ansys.dpf.core.generic_data_container.GenericDataContainer()
+            elif gdc is not None and mesh_info is None:
+                self._gdc = gdc
+            elif gdc is None and MeshInfo is not None:
+                self._gdc = mesh_info._gdc
+        except ValueError:
+            print("Both generic data container and mesh info can't be filled")
+
+    def __call__(self):
+        return self
+
+    def deep_copy(self, server=None):
+        """Create a deep copy of the scoping's data on a given server.
+
+        This method is useful for passiong data from one server instance to another.
+
+        Parameters
+        ----------
+        server : ansys.dpf.core.server, optional
+            Server with the channel connected to the remote or local instance.
+            The default is ``None``, in which case an attempt is made to use the
+            global server.
+
+        Returns
+        -------
+        scoping_copy : Scoping
+        """
+        mesh_info = MeshInfo(server=server)
+        mesh_info._gdc = self._gdc
```

Comment:
You can probably wrap the `_gdc` setter through a setter (see example: https://github.com/ansys/pydpf-core/blob/709032569e01dec991c79ae6ff25c11a862fdbec/src/ansys/dpf/core/custom_type_field.py#L503C1-L517C42)